android/src/main/java/io/sentry/react/RNSentryModule.java

```diff
@@ -414,13 +417,13 @@ public void addBreadcrumb(final ReadableMap breadcrumb) {
             }
 
             if (breadcrumb.hasKey("data")) {
-                ReadableMap data = breadcrumb.getMap("data");
-                ReadableMapKeySetIterator it = data.keySetIterator();
-                while (it.hasNextKey()) {
-                    String key = it.nextKey();
-                    String value = data.getString(key);
-
-                    breadcrumbInstance.setData(key, value);
+                final ReadableMap data = breadcrumb.getMap("data");
+                for(final Map.Entry<String, Object> entry : data.toHashMap().entrySet()) {
+                    final Object value = entry.getValue();
+                    // data is ConcurrentHashMap and can't have null values
+                    if (value != null) {
```

Comment:
It looks good, I'm just wander, it would work the same if we would add the if value != null to the while look, right?

Reply:
nope, `ReadableMapKeySetIterator` does not have a method that returns an `Object`, and previously it was always stringified so it was always `getString()` called.

Reply:
Ohh I see. Thanks.